src/main/scala/dpla/ingestion3/mappers/providers/NaraExtractor.scala

```diff
@@ -0,0 +1,256 @@
+    }
+  }
+
+  def simpleDate(nodeSeq: NodeSeq): Seq[EdmTimeSpan] =
+    nodeSeq.map(node => EdmTimeSpan(originalSourceDate = nodeToDateString(Some(node))))
+
+  def date(xml: NodeSeq): Seq[EdmTimeSpan] = {
+
+    val coverageDates = for {
+      coverageDate <- xml \\ "coverageDates"
+      coverageStartDate = (coverageDate \ "coverageStartDate").headOption
+      coverageEndDate = (coverageDate \ "coverageEndDate").headOption
+      //dateQualifier = coverageDate \ "dateQualifier" //todo not sure what to do with qualifier in EDTF
+      if coverageStartDate.nonEmpty || coverageEndDate.nonEmpty
+
+    } yield {
+      val displayDate = getDisplayDate(coverageStartDate, coverageEndDate)
+      EdmTimeSpan(
+        originalSourceDate = displayDate, //this is cheating?
+        begin = nodeToDateString(coverageStartDate),
+        end = nodeToDateString(coverageEndDate),
+        prefLabel = displayDate
+      )
+    }
+
+    val copyrightDates = simpleDate(xml \\ "copyrightDateArray" \ "proposableQualifiableDate")
+    val productionDates = simpleDate(xml \\ "productionDateArray" \ "proposableQualifiableDate")
+
+    //todo not sure what to do with logicalDate
+    val broadcastDates = simpleDate(xml \\ "broadcastDateArray" \ "proposableQualifiableDate")
+    val releaseDates = simpleDate(xml \\ "releaseDateArray" \ "proposableQualifiableDate")
+
+    Seq(
+      coverageDates,
+      copyrightDates,
+      productionDates,
+      broadcastDates,
+      releaseDates
+    )
+      .collectFirst({ case x => x })
+      .getOrElse(Seq())
+  }
+
+  def publisher(xml: NodeSeq): Seq[String] = {
+
+    val orgs = for {
+      org <- xml \\ "organizationalContributorArray" \ "organizationalContributor"
+      if (org \ "contributorType" \ "termName").text == "Publisher"
+      name <- org \ "termName"
+    } yield name.text
+
+    val persons = for {
+      person <- xml \\ "personalContributorArray" \ "personalContributor"
+      if (person \ "contributorType" \ "termName").text == "Publisher"
+      name <- person \ "termName"
+    } yield name.text
+
+    orgs ++ persons
+  }
+
+  def relation(xml: NodeSeq): Seq[String] = for {
+    parentFileUnit <- xml \\ "parentFileUnit"
+    value1 = (parentFileUnit \ "title").text
+    value2 = (parentFileUnit \ "parentSeries" \ "title").text
+    value3a = (parentFileUnit \ "parentRecordGroup" \ "title").text
+    value3b = (parentFileUnit \ "parentCollection" \ "title").text
+  } yield {
+    val value3 = if (value3a.isEmpty) value3b else value3a
+    //VALUE3"; "VALUE2"; "VALUE1
+    f"""$value3"; $value2"; "$value1""""
+  }
+
+  def rights(xml: NodeSeq): Seq[String] = for {
+    useRestriction <- xml \ "useRestriction"
+    value1 = (useRestriction \ "note").text
+    value2 = (useRestriction \ "specificUseRestrictionArray" \ "specificUseRestriction" \ "termName").text
+    value3 = (useRestriction \ "status" \ "termName").text
+  //VALUE2": "VALUE1" "VALUE3
+  } yield f"""$value2": "$value1" "$value3"""
```

Comment:
They seem to match to me, but maybe I'm taking the notation too literally?

Note that the triple quotes at the beginning and end mean that single quotes no longer require escaping. Also note that I have no idea what's up with the specification in the comment, and I could be implementing it entirely incorrectly.